src/storage-types/src/sources.rs

```diff
@@ -2974,4 +2975,54 @@ mod tests {
             scalar_type_columnar_roundtrip(scalar_type)
         });
     }
+
+    #[mz_ore::test]
+    #[cfg_attr(miri, ignore)] // too slow
+    fn source_proto_serialization() {
```

Comment:
I think this is good enough to finally delete this TODO \o/! https://github.com/MaterializeInc/materialize/blob/7233211be02e41f06b1eea85b4066cfd8d3d0eae/src/repr/src/row/encoding.rs#L1055-L1056